uefi/firmwarevolume.go

```diff
@@ -113,6 +113,19 @@ type FirmwareVolume struct {
 	ExtractPath string
 }
 
+func (n *FirmwareVolume) Apply(v Visitor) error {
+	return v.VisitFV(n)
+}
+
+func (n *FirmwareVolume) ApplyChildren(v Visitor) error {
```

Comment:
Should ApplyChildren also apply to self? Depending on what you choose, document it in the interface docs for ASTNode.

Reply:
I also get the feeling that ApplyChildren could just be a Visitor implementation. E.g.

```go
// RecursiveVisitor is a visitor that recursively applies the nested Visitor to all
// children nodes of each visited node and the visited node itself.
type RecursiveVisitor struct {
  nested Visitor
}

func (rv RecursiveVisitor) VisitFV(fv *FirmwareVolume) error {
  rv.nested.VisitFV(fv)

  for _, f := range fv.Files {
    rv.VisitFile(f)
  }
}

...
```

Reply:
Documented! If I made it call itself, a lot of its uses would turn into infinite recursion. `ApplyChildren` is sort of the default function to call when a visitor skips over a node.